aws/cf.go

```diff
@@ -36,6 +42,47 @@ func (s *Stack) TargetGroupARN() string {
 	return s.targetGroupARN
 }
 
+// IsDeleteInProgress returns true if the stack has already a tag
+// deleteScheduled.
+func (s *Stack) IsDeleteInProgress() bool {
+	if s == nil {
+		return false
+	}
+	_, ok := s.tags[deleteScheduled]
+	return ok
+}
+
+// ShouldDelete returns true if stack is marked to delete and the
+// deleteScheduled tag is after time.Now(). In all other cases it
+// returns false.
+func (s *Stack) ShouldDelete() bool {
+	if s == nil {
```

Comment:
just for nicer Go, you can then have a nil ptr to Stack and you can call ShouldDelete() without checking for nil before